src/zeroconf/_services/browser.py

```diff
@@ -92,23 +95,52 @@
 
 _QuestionWithKnownAnswers = Dict[DNSQuestion, Set[DNSPointer]]
 
+heappop = heapq.heappop
+heappush = heapq.heappush
+
+
+class _ScheduledQuery:
+
+    __slots__ = ('name', 'type_', 'cancelled', 'when_millis')
+
+    def __init__(self, name: str, type_: str, when_millis: float) -> None:
+        """Create a scheduled query."""
+        self.name = name
+        self.type_ = type_
+        self.cancelled = False
+        self.when_millis = when_millis
+
```

Comment:
I think we need all the equality methods here.  Check what cpython does for timer handles